Clarify how to manually add LIFX bulbs and how to calculate zones (#24663)
home-assistant/home-assistant.github.io: source/_integrations/lifx.markdown

ha_dhcp: true
---

> The LIFX integration automatically discovers [LIFX](https://www.lifx.com) lights on each network that is enabled in Home Assistant's [network configuration](/integrations/network). If any of your LIFX lights are not automatically discovered, you can add them manually using the user interface by following the configuration steps below for each light you want to add:

Reviewer: ```suggestion
The LIFX integration automatically discovers [LIFX](https://www.lifx.com) lights on each network that is enabled in Home Assistant's [network configuration](/integrations/network). Suppose any of your LIFX lights are not automatically discovered. In that case, you can add them manually using the user interface by following the configuration steps below for each light you want to add:
```

{% include integrations/config_flow.md %}

## Set state

> LIFX lights allow a change of color and brightness even when they are turned off. This way you can control the light during the day so its settings are correct when events for turning on are received, for example from motion detectors or external buttons.

Reviewer: ```suggestion
LIFX lights allow a change of color and brightness even when they are turned off. This way, you can control the light during the day, so its settings are correct when events for turning on are received, for example, from motion detectors or external buttons.
```

| ---------------------- | ----------- |
| `entity_id` | String or list of strings that point at `entity_id`s of lights. Use `entity_id: all` to target all.
| `transition` | Duration (in seconds) for the light to fade to the new state.
| `zones` | List of integers for the zone numbers to affect. See **Calculating zones to affect** below for more detail.
| `power` | Turn the light on (`True`) or off (`False`). Leave out to keep the power as it is.
| `...` | Use `color_name`, `brightness` etc. from [`light.turn_on`](/integrations/light/#service-lightturn_on) to specify the new state.

#### Calculating zones to affect

> The LIFX Z and LIFX Lightstrip have 8 zones per segment and a maximum allowed length of 10 segments, which results in a maximum zone count of 80.

Reviewer: ```suggestion
The LIFX Z and LIFX Lightstrip have eight (8) zones per segment and a maximum allowed length of 10, resulting in a maximum zone count of 80.
```

> The LIFX Beam has 10 zones in each beam and 1 zone in each corner. You can add a maximum of 8 beams and 2 corners to a controller, which results in a maximum zone count of 82.

Reviewer: ```suggestion
The LIFX Beam has ten (10) zones in each beam and 1 zone in each corner. You can add a maximum of 8 beams and two (2) corners to a controller, which results in a maximum zone count of 82.
```

> All devices start counting zones at 0 which means the zone numbers for the `zones` attribute above range from 0 to 79 for the LIFX Z and Lightstrip and 0 to 81 for the LIFX Beam.

Reviewer: ```suggestion
All devices start counting zones at zero (0), which means the zone numbers for the `zones` attribute above range from 0 to 79 for the LIFX Z and Lightstrip and 0 to 81 for the LIFX Beam.
```
